GVFS/GVFS.Common/NuGetUpgrade/NuGetUpgrader.cs

```diff
@@ -597,6 +591,13 @@ private bool EnsureNuGetFeedInitialized(out string error)
         {
             if (!this.isNuGetFeedInitialized)
             {
+                Debug.Assert(this.credentialStore != null, "Attempting to access nuget feed without a credential store configured.");

```

Comment:
I don't think we have been using `Debug.Assert` in the code base up to this point.  What value does this bring?  What will the user see when they hit this situation?

Reply:
> I don't think we have been using Debug.Assert in the code base up to this point. What value does this bring? What will the user see when they hit this situation?

I could throw an exception here instead... Users should never see this, as it indicates a programming error. If they do see it, the will see the error from the line below.

I guess the question is - do we want to throw in the release builds, or throw in just debug builds and have this error message propagated in release builds...

Reply:
>I could throw an exception here instead... Users should never see this, as it indicates a programming error. If they do see it, the will see the error from the line below.
>
>I guess the question is - do we want to throw in the release builds, or throw in just debug builds and have this error message propagated in release builds...

Elsewhere in the code that we've added checks like this (i.e. invalid states that indicate programmer error) we've thrown an `InvalidOperationException`.  I think the same would be appropriate here.

Reply:
@jamill actually, is it possible to create a `NuGetUpgrader` without using one of the methods/constructors that takes a `credentialStore` as a parameter?

If not, could those methods be updated to throw an `ArgumentNullException` if `credentialStore` is null rather than checking if `this.credentialStore` is null here?

Reply:
> is it possible to create a NuGetUpgrader without using one of the methods/constructors that takes a credentialStore as a parameter?

With the current design, this is possible (unfortunately). Instead of refactoring to remove this behavior at this time, I was going to keep this bit of awkwardness and will throw an exception if we call into this incorrectly...